packages/spector/src/actions/server-test.ts

```diff
@@ -194,42 +142,61 @@ export async function serverTest(scenariosPath: string, options: ServerTestOptio
       const obj: ServerTestsGenerator = new ServerTestsGenerator(name, api, baseUrl);
       try {
         await obj.executeScenario();
-        success_diagnostics.push({
-          scenario_name: name,
-          status: "success",
-          message: "executed successfully",
+        successfullScenarios.push({
+          name,
+          pathlikeName,
         });
       } catch (e: any) {
-        failure_diagnostics.push({
-          scenario_name: name,
-          status: "failure",
-          message: `code = ${e.code} \n message = ${e.message} \n name = ${e.name} \n stack = ${e.stack} \n status = ${e.status}`,
-        });
+        if (e instanceof ValidationError) {
+          failureDiagnostics.push({
+            scenarioName: name,
+            message: [
+              `Validation failed: ${e.message}:`,
+              ` Expected:\n    ${inspect(e.expected)}`,
+              ` Actual:\n    ${inspect(e.actual)}`,
+            ].join("\n"),
+          });
+        } else {
+          failureDiagnostics.push({
+            scenarioName: name,
+            message: `code = ${e.code} \n message = ${e.message} \n name = ${e.name} \n stack = ${e.stack} \n status = ${e.status}`,
+          });
+        }
       }
     }
   }
 
   // 4. Print diagnostics
-  logger.info("Server Tests Diagnostics Summary");
+  log("");
+  log("Server Tests Diagnostics Summary");
 
-  // Exit with error if no diagnostics were ran
-  if (success_diagnostics.length === 0 && failure_diagnostics.length === 0) {
+  if (successfullScenarios.length === 0 && failureDiagnostics.length === 0) {
     logger.error("No scenarios were executed");
     process.exit(-1);
   }
 
-  if (success_diagnostics.length > 0) logger.info("Success Scenarios");
-  success_diagnostics.forEach((diagnostic) => {
-    logger.info(`${pc.green("✓")} Scenario: ${diagnostic.scenario_name} - ${diagnostic.message}`);
+  if (successfullScenarios.length > 0) log("Success Scenarios");
+  successfullScenarios.forEach((diagnostic) => {
+    log(`${pc.green("✓")} Scenario: ${pc.cyan(diagnostic.name)}`);
   });
 
-  if (failure_diagnostics.length > 0) logger.error("Failure Scenarios");
-  if (failure_diagnostics.length > 0) {
-    logger.error("Failed Scenario details");
-    failure_diagnostics.forEach((diagnostic) => {
-      logger.error(`${pc.red("✘")} Scenario: ${diagnostic.scenario_name}`);
-      logger.error(`${diagnostic.message}`);
+  if (failureDiagnostics.length > 0) log("Failure Scenarios");
+  if (failureDiagnostics.length > 0) {
+    log("Failed Scenario details");
```

Comment:
I know you probably just renamed this variable but these consecutive if statements stand out :smile:

Reply:
yeah removed the one above